src/benchmarks/pmembench.cpp

```diff
@@ -172,7 +174,7 @@ static struct version_s {
 static struct bench_list benchmarks;
 
 /* common arguments for benchmarks */
-static struct benchmark_clo pmembench_clos[11];
+static struct benchmark_clo pmembench_clos[12];
```

Comment:
change this to a `std::vector<struct benchmark_clo> pmembench_clos;`
and change the entries to `pmembench_clos.emplace_back("main_affinity", "Set affinity...", CLO_TYPE_INT);`

Reply:
IMO it will won't work without other changes, but if it does, it would look terrible:
pmembench_clos.emplace_back(NULL, "main_affinity", "Set affinity...", CLO_TYPE_INT, clo_field_offset(struct benchmark_args, main_affinity), "-1", false, (clo_field_size(struct benchmark_args, main_affinity), CLO_INT_BASE_DEC, -1, LONG_MAX), NULL, NULL);

Rewriting benchmarks to c++ is something what we want to do in future but it is not a part of this patch

Reply:
Looks legit 😈